src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -123,10 +148,15 @@ private void Execute(Transaction transaction, BlockHeader block, ITxTracer txTra
             
             // restore is CallAndRestore - previous call, we will restore state after the execution
             bool restore = (executionOptions & ExecutionOptions.Restore) != ExecutionOptions.None;
+            
             // commit - is for standard execute, we will commit thee state after execution 
             bool commit = (executionOptions & ExecutionOptions.Commit) != ExecutionOptions.None || eip658NotEnabled;
             //!commit - is for build up during block production, we won't commit state after each transaction to support rollbacks
             //we commit only after all block is constructed 
+            
+            // commitAndRestoreWithoutZeroGasPrice is CallAndRestore without a zero gas price
+            bool isCommitAndRestoreWithoutZeroGasPrice = commit && restore && ((executionOptions & ExecutionOptions.ZeroGasPrice) == ExecutionOptions.None);
```

Comment:
Can we simplify it in any way?

Reply:
It seems to me that it looks more understandable in this form.